tests/sub_port_interfaces/conftest.py

```diff
@@ -46,6 +46,18 @@ def pytest_addoption(parser):
     )
 
 
+@pytest.fixture(scope='module', autouse=True)
+def skip_unsupported_asic_type(duthost):
+    SUBPORT_UNSUPPORTED_ASIC_LIST = ["th2"]
+    vendor = duthost.facts["asic_type"]
+    hostvars = duthost.host.options['variable_manager']._hostvars[duthost.hostname]
```

Comment:
There is a tool for getting host vars: https://github.com/Azure/sonic-mgmt/blob/master/tests/common/utilities.py#L264
Caching is enabled for function `get_host_visible_vars`. Can you consider using this function?